### ISSUE TITLE
Log4j2Metrics inaccurate with async appender

### ISSUE THREAD

**Issue by ksambhav** [micrometer-metrics/micrometer#3980]
io.micrometer.core.instrument.binder.logging.Log4j2Metrics  reports inaccurate metrics in case of async appender.
The decision to increment the counter is done by method :-

```
        @Override
        public Result filter(LogEvent event) {

            if (!isAsyncLogger || isAsyncLoggerAndEndOfBatch(event)) {
                incrementCounter(event); // always increments the counter by 1
            }

            return Result.NEUTRAL;
        }

```

When `isAsyncLoggerAndEndOfBatch(event)` is true, counter should get incremented by "batch" size. (Not sure if its possible to get it from `LogEvent` parameter. 

**Environment**
PROD
 - Micrometer version : 1.11.1
 - Micrometer registry : Any. I tried with AzureMetricRegistry and SimpleRegistry
 - OS: Linux
 - Java version: 17

**To Reproduce**
How to reproduce the bug:
Unzip and run attached Spring Boot application. It prints 2000 error level logs in burst every 30 secs.
Counter can be accessed using http://localhost:8080/actuator/metrics/log4j2.events

Repeat again with sync appender by change `<AsyncRoot>` to simply `<Root>` in `log4j2.xml`.  This time counter value will be correct this time.

**Expected behavior**
Counter should report accurate metrics in case of async appender

**Additional context**
https://stackoverflow.com/questions/76660558/log4j2-metrics-with-micrometer
[Archive.zip](https://github.com/micrometer-metrics/micrometer/files/12025780/Archive.zip)

**Comment by shakuzen** [comment]
It sounds like this is a duplicate of https://github.com/micrometer-metrics/micrometer/issues/2176. Let me know if you disagree and why. We can reopen if it is a separate issue.

**Comment by ksambhav** [comment]
If the fix for #2176 will also cover case when counter is increment by the "batch size" instead just 1, we can keep this one closed. In the end we just want accurate count anyways.
